src/Databases/DatabaseOnDisk.cpp

```diff
@@ -67,6 +68,11 @@ std::pair<String, StoragePtr> createTableFromAST(
     ast_create_query.attach = true;
     ast_create_query.setDatabase(database_name);
 
+    if (!ast_create_query.sql_security && ast_create_query.supportSQLSecurity() && !context->getServerSettings().ignore_empty_sql_security_in_create_view_query)
```

Comment:
This place seems to be too generic for that kind of intervention. Perhaps you can move this code to the constructor of `StorageMaterializedView`?